src/routes/NewBoard/NewBoard.tsx

```diff
@@ -19,8 +22,109 @@ export const NewBoard = () => {
   const [accessPolicy, setAccessPolicy] = useState(0);
   const [passphrase, setPassphrase] = useState("");
   const [extendedConfiguration, setExtendedConfiguration] = useState(false);
+  const [importFile, setImportFileConfiguration] = useState(false);
+  const [showPasswordModal, setShowPasswordModal] = useState(false);
+  const [loadedFile, setFile] = useState<File>();
+  const [completeBoard, setImportBoard] = useState<BoardContent>();
+  const fileInputRef = useRef<HTMLInputElement>(null);
+
+  type ImportBoardRequest = {
+    name: string;
+    description?: string;
+    accessPolicy: string;
+    passphrase?: string;
+  };
+
+  type BoardContent = {
+    board: ImportBoardRequest;
+    columns: Column[];
+    notes: Note[];
+    participants: Participant;
+    voting: Voting;
+  };
+
+  const handleDragOver = (event: React.DragEvent<HTMLDivElement>) => {
+    event.preventDefault();
+  };
+
+  const handleDragLeave = (event: React.DragEvent<HTMLDivElement>) => {
+    event.preventDefault();
+  };
+
+  const closeModal = () => {
+    if (fileInputRef.current) {
+      fileInputRef.current.value = "";
+    }
+
+    setShowPasswordModal(false);
+    setPassphrase("");
+  };
+
+  const handleFileEvent = (event: React.ChangeEvent<HTMLInputElement> | React.DragEvent<HTMLDivElement>) => {
+    let file: File | null = null;
+    if ("dataTransfer" in event) {
+      event.preventDefault();
+      file = event.dataTransfer?.files?.[0] || null;
+    } else if ("target" in event) {
+      file = event.target.files?.[0] || null;
+    }
+
+    if (!file) return;
+
+    const reader = new FileReader();
+    reader.onload = (e) => {
+      const content = e.target?.result as string;
 
-  async function onCreateBoard() {
+      try {
+        const data = JSON.parse(content) as BoardContent;
+        const board: BoardContent = {
+          board: {
+            name: data.board.name,
+            description: data.board.description,
+            accessPolicy: data.board.accessPolicy,
+          },
+          columns: data.columns,
+          notes: data.notes,
+          participants: data.participants,
+          voting: data.voting,
+        };
+        if (board.board.accessPolicy === "BY_PASSPHRASE") {
+          setShowPasswordModal(true);
+          setAccessPolicy(1);
+        }
+        setImportBoard(board);
+      } catch (error) {
+        // console.error("Error parsing JSON:", error);
+        // TODO toast error instead
+      }
+    };
+    setFile(file);
+    reader.readAsText(file);
+  };
+
+  const onImportBoard = async () => {
+    if (completeBoard && accessPolicy === AccessPolicy.BY_PASSPHRASE && passphrase) {
+      completeBoard.board.passphrase = passphrase;
+    }
+
+    const boardId = await API.importBoard(JSON.stringify(completeBoard));
```

Comment:
direct API calls like that are not in the spirit of our architecture. Please use a thunk for this